src/chains/tests/test_signals_ff_hook_events.py

```diff
@@ -0,0 +1,376 @@
+        assert (
+            responses.calls[3].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+
+@override_settings(
+    FF_HOOK_EVENTS=True,
+    CGW_URL="http://127.0.0.1",
+    CGW_FLUSH_TOKEN="example-token",
+)
+class WalletHookTestCase(TestCase):
+    @responses.activate
+    def test_on_wallet_create_with_no_chain(self) -> None:
+        Wallet(key="Test Wallet").save()
+
+        # Creating a wallet with no chains should not trigger any webhook
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_wallet_create_with_chain(self) -> None:
+        chain = ChainFactory.create()
+        WalletFactory.create(key="Test Wallet", chains=(chain,))
+
+        # 1 call for Chain creation, 1 call for Wallet creation, 1 call for M2M update
+        assert len(responses.calls) == 3
+        assert isinstance(responses.calls[2], responses.Call)
+        assert responses.calls[
+            2
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain.id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[2].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[2].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_wallet_delete_with_no_chain(self) -> None:
+        wallet = Wallet(key="Test Wallet")
+
+        wallet.save()  # create
+        wallet.delete()  # delete
+
+        # deleting a wallet with no chains should not trigger any webhook
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_wallet_delete_with_chain(self) -> None:
+        chain = ChainFactory.create()
+        wallet = WalletFactory.create(key="Test Wallet", chains=(chain,))
+
+        wallet.delete()
+
+        # 1 call for Chain creation, 1 call for Wallet creation, 1 call for M2M update, 1 call for Wallet deletion
+        assert len(responses.calls) == 4
+        assert isinstance(responses.calls[3], responses.Call)
+        assert responses.calls[
+            3
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain.id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[3].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[3].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_wallet_update_with_no_chain(self) -> None:
+        wallet = Wallet(key="Test Wallet")
+
+        wallet.save()  # create
+        wallet.key = "Test Wallet v2"
+        wallet.save()  # update
+
+        # Updating a wallet with no chains should not trigger any webhook
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_feature_update_with_chain(self) -> None:
```

Comment:
Is this test name accurate? It seems `feature` is used in the name but the test is referred to `wallet`, right?

Reply:
Addressed in [bfcbbbf](https://github.com/safe-global/safe-config-service/pull/964/commits/bfcbbbfc65b52d790114b3960c1e3dce3b2e30be)